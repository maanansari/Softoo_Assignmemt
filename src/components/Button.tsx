import React from 'react';
import {Text, TouchableOpacity} from 'react-native';
import {style} from './style';

type Props = {
  title: string;
  onPress: () => void;
};

const Button: React.FC<Props> = ({title, onPress}) => {
  return (
    <TouchableOpacity testID="btnContainer" onPress={onPress} style={style.buttonStyle}>
      <Text testID="btnText" style={style.buttonTextStyle}>
        {title}
      </Text>
    </TouchableOpacity>
  );
};

export default Button;
