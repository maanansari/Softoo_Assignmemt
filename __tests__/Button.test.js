import Button from '../src/components/Button';
import {fireEvent, render} from '@testing-library/react-native';

const props = {
  onPress: jest.fn(),
  title: '',
};

describe('Button', () => {
  let props = {
    title: 'Submit',
    onPress: jest.fn(),
  };

  it('match snap', () => {
    const {toJSON} = render(<Button {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('match snap', () => {
    const {getByTestId} = render(<Button {...props} />);
    const buttonContainer = getByTestId('btnContainer');
    const buttonText = getByTestId('btnText');
    expect(buttonContainer).toBeDefined();
    expect(buttonText).toBeDefined();
    expect(buttonText.props.children).toBe(props.title);

    fireEvent.press(buttonContainer);
    expect(props.onPress).toHaveBeenCalled();
  });
});
