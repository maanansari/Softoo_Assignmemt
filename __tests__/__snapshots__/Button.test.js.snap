// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button match snap 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "center",
      "backgroundColor": "#0F52BA",
      "borderRadius": 5,
      "marginHorizontal": 4,
      "marginVertical": 2,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 7,
    }
  }
  testID="btnContainer"
>
  <Text
    style={
      {
        "color": "#FFF",
        "fontSize": 17,
      }
    }
    testID="btnText"
  >
    Submit
  </Text>
</View>
`;
